### PR TITLE
Remove IsDFluxnodeActive calls

### PR DESCRIPTION
**Pull request by w2vy**
Added runtime error if not fFluxnode for good measure, other RPC calls use it too

In non-rpc functions this state was checked directly by block number when examining blocks of any age

### CODE REVIEW

File: src/rpc/zelnode.cpp [hunk 1]
```diff
@@ -1126,49 +1115,46 @@ UniValue getfluxnodecount (const UniValue& params, bool fHelp, string cmdname)
 
     UniValue obj(UniValue::VOBJ);
 
-    if (IsDFluxnodeActive())
+    if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (TheTrunk) [inline src/rpc/zelnode.cpp:1118]: This rpc can be run on a node that is not a fluxnode

File: src/rpc/zelnode.cpp [hunk 2]
```diff
@@ -1197,50 +1183,47 @@ UniValue getmigrationcount (const UniValue& params, bool fHelp)
                 "\nExamples:\n" +
                 HelpExampleCli("getmigrationcount", "") + HelpExampleRpc("getmigrationcount", ""));
 
-    if (IsDFluxnodeActive())
+    if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (TheTrunk) [inline src/rpc/zelnode.cpp:1186]: This rpc can be run on a node that is not a fluxnode

File: src/rpc/zelnode.cpp [hunk 3]
```diff
@@ -1068,32 +1059,30 @@ UniValue fluxnodecurrentwinner (const UniValue& params, bool fHelp, string cmdna
...
-                obj.pushKV("added_height", data.nAddedBlockHeight);
-                obj.pushKV("confirmed_height", data.nConfirmedBlockHeight);
-                obj.pushKV("last_confirmed_height", data.nLastConfirmedBlockHeight);
-                obj.pushKV("last_paid_height", data.nLastPaidHeight);
-                obj.pushKV("tier", TierToString(data.nTier));
-                obj.pushKV("payment_address", EncodeDestination(dest));
-                ret.pushKV(strWinner, obj);
-            }
-        }
+    if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (TheTrunk) [inline src/rpc/zelnode.cpp:1062]: This rpc can be run on a node that is not a fluxnode

File: src/rpc/zelnode.cpp [hunk 4]
```diff
@@ -908,55 +906,53 @@ UniValue getstartlist(const UniValue& params, bool fHelp)
                 "\nExamples:\n" +
                 HelpExampleCli("getstartlist", "") + HelpExampleRpc("getstartlist", ""));
 
-    if (IsDFluxnodeActive()) {
-        UniValue wholelist(UniValue::VARR);
+    if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (TheTrunk) [inline src/rpc/zelnode.cpp:909]: This rpc can be run on a node that is not a fluxnode

File: src/rpc/zelnode.cpp [hunk 5]
```diff
@@ -837,54 +837,52 @@ UniValue getdoslist(const UniValue& params, bool fHelp)
                 "\nExamples:\n" +
                 HelpExampleCli("getdoslist", "") + HelpExampleRpc("getdoslist", ""));
 
-    if (IsDFluxnodeActive()) {
-        UniValue wholelist(UniValue::VARR);
+    if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (TheTrunk) [inline src/rpc/zelnode.cpp:840]: This rpc can be run on a node that is not a fluxnode

**Review by TheTrunk — CHANGES_REQUESTED** [review]

File: src/rpc/zelnode.cpp [hunk 6]
```diff
@@ -990,53 +986,48 @@ UniValue getfluxnodestatus (const UniValue& params, bool fHelp, string cmdname)
 
     if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (w2vy) [inline src/rpc/zelnode.cpp:987]: I think this should be deleted too, this overloaded function is called by getzelnodestatus

File: src/rpc/zelnode.cpp [hunk 7]
```diff
@@ -1289,63 +1272,62 @@ UniValue listfluxnodeconf (const UniValue& params, bool fHelp, string cmdname)
 
     UniValue ret(UniValue::VARR);
 
+    if (!fFluxnode) throw runtime_error("This is not a Flux Node");
```
Comment (w2vy) [inline src/rpc/zelnode.cpp:1275]: Overloaded function called by listfluxnodeconf and listzelnodeconf, only throw error from listfluxnodeconf wrapper?

File: src/rpc/zelnode.cpp (hunk 5 above)
Comment (w2vy) [reply src/rpc/zelnode.cpp:840]: I am taking a fresh look, I want to review the code from before I made changes

File: src/rpc/zelnode.cpp (hunk 6 above)
Comment (w2vy) [reply src/rpc/zelnode.cpp:987]: I reviewed against code before I 'fixed' anything and yes this stays, just wanted to make sure I didn't may a mistake earlier.

File: src/rpc/zelnode.cpp (hunk 7 above)
Comment (w2vy) [reply src/rpc/zelnode.cpp:1275]: Deleted this one too, original code did not have it

File: src/rpc/zelnode.cpp (hunk 6 above)
Comment (TheTrunk) [reply src/rpc/zelnode.cpp:984]: Yes you are correct getfluxnodestatus/getzelnodestatus can only be run on flux node 👍

**Review by TheTrunk — APPROVED** [review]

**Review by blondfrogs — APPROVED** [review]

**Pull request merged**